linera-execution/tests/test_execution.rs

```diff
@@ -64,289 +68,92 @@ async fn test_missing_bytecode_for_user_application() -> anyhow::Result<()> {
     Ok(())
 }
 
-#[derive(Clone)]
-struct TestModule<const IS_CALLER: bool> {
-    owner: Owner,
-}
-
-struct TestApplication<const IS_CALLER: bool, Runtime> {
-    owner: Owner,
-    runtime: Runtime,
-}
-
-impl<const IS_CALLER: bool> TestModule<IS_CALLER> {
-    fn new(owner: Owner) -> Self {
-        Self { owner }
-    }
-}
-
-/// A fake operation for the [`TestApplication`] which can be easily "serialized" into a single
-/// byte.
-#[repr(u8)]
-enum TestOperation {
-    Completely,
-    LeakingSession,
-    FailingCrossApplicationCall,
-}
-
-impl UserContractModule for TestModule<true> {
-    fn instantiate(
-        &self,
-        runtime: ContractSyncRuntime,
-    ) -> Result<Box<dyn UserContract + Send + Sync + 'static>, ExecutionError> {
-        Ok(Box::new(TestApplication::<true, ContractSyncRuntime> {
-            owner: self.owner,
-            runtime,
-        }))
-    }
-}
-
-impl UserContractModule for TestModule<false> {
-    fn instantiate(
-        &self,
-        runtime: ContractSyncRuntime,
-    ) -> Result<Box<dyn UserContract + Send + Sync + 'static>, ExecutionError> {
-        Ok(Box::new(TestApplication::<false, ContractSyncRuntime> {
-            owner: self.owner,
-            runtime,
-        }))
-    }
-}
-
-/// Key to store the application ID to call.
-static CALLEE_ID_KEY: [u8; 1] = [0];
-/// Key to store some dummy data in the application state.
-static DUMMY_STATE_KEY: [u8; 1] = [1];
+#[tokio::test]
+async fn test_simple_user_operation() -> anyhow::Result<()> {
```

Comment:
This test now seems to be testing a few different things, including cross-application calls, authentication forwarding, session calls, and runtime storage. I'm thinking about adding a `TODO` here to split this test into smaller and more specialized tests. But I'd like to know your opinion about this and the overall refactor before opening the issue :sweat_smile:

Reply:
Yes, I would suggest adding a TODO pointing a GH issue labeled as a "good first task".